dbConnectivityChecker.js

```diff
@@ -0,0 +1,44 @@
+const MongoClient = require('mongodb').MongoClient;
+
+class DbConnectivityChecker {
+    constructor(io) {
+        this.url = process.env.DATABASE || 'mongodb://0.0.0.0:27017';
+        this.io = io;
+    }
+
+    async emitDbAvailable() {
+        if (this.io) {
+            this.io.emit('dbConnectivityState', { dbState: true });
+        } else {
+            console.error("Socket.io instance not available.");
+        }
+    }
+
+    async emitDbUnAvailable() {
+        if (this.io) {
+            this.io.emit('dbConnectivityState', { dbState: false });
+        } else {
+            console.error("Socket.io instance not available.");
+        }
+    }
+
+    async checkDbConnectivity( dbCheckInterval ) {
+        let heartBeatInterval = (dbCheckInterval * 1000);
+        try {
+            const clientInit = new MongoClient(this.url, { useNewUrlParser: true, useUnifiedTopology: true, 
+                serverSelectionTimeoutMS: 4000, heartbeatFrequencyMS: heartBeatInterval });
+            
+            clientInit.on ("serverHeartbeatSucceeded",() => { this.emitDbAvailable() });
```

Comment:
Emitting db available every 2 seconds may not be required. We can optimise on this, the way it was done in earlier dbPing funciton.